### ISSUE TITLE
Bootstrap buffer doesn't provide speedup

### ISSUE THREAD

**Issue by klauspost** [pquerna/ffjson#61]
In both your test and in a more complex one, I get better performance when setting the size of the bootstrap buffer to 0 bytes:

```
BenchmarkComplexMarshalJSONNative         100000             17252 ns/op  76.80 MB/s
BenchmarkMarshalJSONNative       1000000              2364 ns/op          41.87MB/s
```

With bootstrap size set to 0:
```
BenchmarkComplexMarshalJSONNative         100000             16767 ns/op  79.02 MB/s
BenchmarkMarshalJSONNative       1000000              2104 ns/op          47.04 MB/s
```
The difference is pretty consistent on my machine. How does it look on yours? I would expect the slowdown to come from zeroing the array and in some cases copying the slice into a bigger one.

I have tried setting the size to both lower and higher values, and lower gives speedup, while higher gives slowdown. '0' was the fastest value.

**Comment by klauspost** [comment]
If bootstrap is removed, Truncate should probably be changed to:

**Comment by pquerna** [comment]
On `sync.Pool`: We don't have a multi-go-routine benchmark right now -- I'd like to add one before getting too "into" it -- in a real world web server, that is processing requests in multiple go-routines, today there is zero locking from ffjson -- reusing the Buffer might be worth it, but i'd like to be able to test it more realistically before adding usage of `sync.Pool`.

**Comment by klauspost** [comment]
I tried a few approaches for buffering, but the only way I found worked was if it was the only major speedup would be if the byte buffer was re-used. The problem of course is that we are returning the byte array to the json encoder/the user.

The caller can still do the buffering like this:
```Go
var buffers = sync.Pool{New: func() interface{} { return &fflib.Buffer{} }}

func handle() {
	var b = buffers.Get().(*fflib.Buffer)
	err := record.MarshalJSONBuf(b)
	json := b.Bytes()
	// Finish working with the bytes, so the buffer is no longer needed.
	b.Reset()
	buffers.Put(b)
}
```
This works fine if the bootstrap is removed and the truncate function is modified as described above.

I can send you a PR, if you'd like to see it.
